src/worker.js

```diff
@@ -156,7 +156,78 @@ self.freeTrack = () => {
  * @param {!string} url the URL of the subtitle file.
  */
 self.setTrackByUrl = ({ url }) => {
-  self.setTrack({ content: read_(url) })
+  // simulate throttling due to network or resource pressure
+  // let delay = 0;
+  let fontnameIndex;
+
+  const process = line => {
+    preprocess: if (availableFonts) {
+      // Ignore empty lines and comments
+      if (!line || line.startsWith(';')) {
+        break preprocess;
+      }
+
+      // Reset format with new sections
+      if (line.startsWith('[')) {
+        fontnameIndex = undefined;
+        break preprocess;
+      }
+
+      const verb = line.split(':')[0];
+
+      if (verb === 'Format') {
+        fontnameIndex = line.split(',').findIndex(s => s.trim() === 'Fontname');
+      } else if (verb === 'Style') {
+        let font = line.split(',')[fontnameIndex].trim();
+
+        if (font) {
+          findAvailableFonts(font);
+        }
+      }
+
+      const regex = /\\fn([^\\}]*?)[\\}]/g;
+      let matches;
+      while ((matches = regex.exec(line)) !== null) {
+        findAvailableFonts(matches[1])
+      }
+    }
+
+    // delay += 200 + Math.random() * 500;
+    // setTimeout(() => {
+    if (dropAllBlur) line = dropBlur(line)
+    jassubObj.processLine(line);
+    // }, delay);
+  };
+
+  _fetch(url).then(response => {
+    let partialLine = '';
+
+    return response.body
+      .pipeThrough(new TextDecoderStream())
+      .pipeTo(new WritableStream({
+        start: () => {
+          jassubObj.newEmptyTrack();
+          subtitleColorSpace = libassYCbCrMap[jassubObj.trackColorSpace];
+          postMessage({ target: 'verifyColorSpace', subtitleColorSpace });
+        },
+        write: (chunk) => {
+          // Extract lines from chunk
+          const lines = (partialLine + chunk).split(/\r\n|[\r\n]/g);
+
+          // Save last line, as it might be incomplete
+          partialLine = lines.pop() || '';
+
+          // Process each complete line
+          lines.forEach(process);
```

Comment:
no. absolutely not. never ever do this for functions like map, foreach, reduce etc

```js
for (const line of lines) {
  process(line)
}
```